modApiExt/weapon.lua

```diff
@@ -22,6 +22,55 @@ function weapon:plusTarget(center, size)
 	return ret
 end
 
+--Returns all the varieties of the past weapon name that are defined
+function weapon:getAllExistingNamesForWeapon(weaponBaseName)
+	local allExisting = {}
+	for _, possiblility in pairs({weaponBaseName, weaponBaseName.."_A", weaponBaseName.."_B", weaponBaseName.."_AB"}) do
+		if _G[possiblility] then
+			table.insert(allExisting, possiblility)
+		end
+	end
+	
+	return allExisting
+end
+
+--Returns the upgrade suffix of the weapon i.e. _A,_B,_AB, or empty
```

Comment:
In commit 831da06 there's been a fix for weapon upgrade detection, please apply the same changes here.

For convenience, here's what you need to replace the old `getUpgradeSuffix` function with:

```lua
local function isPowered(upgrade)
	return upgrade and (#upgrade == 0 or (upgrade[1] and upgrade[1] > 0))
end

function weapon:getUpgradeSuffix(wtable)
	local hasUpgradeA = isPowered(wtable.upgrade1)
	local hasUpgradeB = isPowered(wtable.upgrade2)

	if hasUpgradeA and hasUpgradeB then
		return "_AB"
	elseif hasUpgradeA then
		return "_A"
	elseif hasUpgradeB then
		return "_B"
	end

	return ""
end
```
After this change, the conflict with `pawn.lua` can be resolved by just deleting the conflicting lines.